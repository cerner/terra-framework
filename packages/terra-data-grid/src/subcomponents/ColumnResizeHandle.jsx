import React, {
  useContext, useRef, useEffect, useState,
} from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import * as KeyCode from 'keycode-js';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import ColumnContext from '../utils/ColumnContext';
import styles from './ColumnResizeHandle.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The cell's column position in the grid. This is zero based.
   */
  columnIndex: PropTypes.number,
  /**
   * Text of the column associated with the divider.
   */
  columnText: PropTypes.string.isRequired,
  /**
   * Number that specifies the Width of the associated column in pixels.
   */
  columnWidth: PropTypes.number.isRequired,
  /**
    * Number that specifies the height of the resize handle in pixels.
    */
  height: PropTypes.number.isRequired,
  /**
   * Numeric increment in pixels to adjust column width when resizing via the keyboard.
   */
  columnResizeIncrement: PropTypes.number,
  /**
   * Control is the active element
   */
  isActive: PropTypes.bool,
  /**
   * Handler function to update isActive for parent.
   */
  setIsActive: PropTypes.func,
  /**
   * Number that specifies the minimum column width in pixels.
   */
  minimumWidth: PropTypes.number.isRequired,
  /**
   * Number that specifies the maximum column width in pixels.
   */
  maximumWidth: PropTypes.number.isRequired,
  /**
   * Function that is called when onMouseDown event is triggered for the resize handle.
   */
  onResizeMouseDown: PropTypes.func.isRequired,
  /**
   * Function that is called when onMouseDown event is triggered for the resize handle.
   */
  onResizeMouseUp: PropTypes.func.isRequired,
  /**
   * Function that is called when the the keyboard is used to adjust the column size.
   */
  onResizeHandleChange: PropTypes.func,
  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
};

const defaultProps = {
  columnResizeIncrement: 10,
};

const ColumnResizeHandle = (props) => {
  const {
    columnIndex,
    columnResizeIncrement,
    columnText,
    columnWidth,
    height,
    intl,
    isActive,
    maximumWidth,
    minimumWidth,
    onResizeHandleChange,
    onResizeMouseDown,
    onResizeMouseUp,
    setIsActive,
  } = props;

  // Retrieve current theme from context
  const theme = useContext(ThemeContext);
  const columnContext = useContext(ColumnContext);

  // Ref variable for native resize handle element
  const resizeHandleRef = useRef();

  const [isAriaLabel, setIsAriaLabel] = useState(false);
  const [isAriaValueText, setIsAriaValueText] = useState(false);
  const [isNavigationEnabled, setNavigationEnabled] = useState(true);

  useEffect(() => {
    if (isActive) {
      resizeHandleRef.current.focus();
    }
  }, [isActive]);

  // Mouse down event listener to give focus to resize handler and notify the provider
  const onMouseDown = (event) => {
    // Set focus to resize handle DOM element
    resizeHandleRef.current.focus();

    // Execute callback function to notify consumer of mouse down event
    onResizeMouseDown(event);

    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
    // Prevent default dragging behavior
    event.preventDefault();
  };

  const onMouseUp = () => {
    onResizeMouseUp();
  };

  const fitToTable = () => {
    // Update resize handle height to match parent table height
    resizeHandleRef.current.style.height = `${height}px`;

    // Assistive technologies should announce aria-label text once focused
    setIsAriaLabel(true);
  };

  const onMouseLeave = () => {
    if (document.activeElement !== resizeHandleRef.current) {
      resizeHandleRef.current.style.height = '100%';
    }
  };

  // Handle column resize handle keyboard navigation
  const onKeyDown = (event) => {
    const key = event.keyCode;
    switch (key) {
      case KeyCode.KEY_SPACE:
      case KeyCode.KEY_RETURN:
        setNavigationEnabled(false);
        // Lock focus into component
        resizeHandleRef.current.focus();

        // Assistive technologies should avoid announcing aria-label while focus locked, but announce aria-valueText instead
        setIsAriaLabel(false);
        setIsAriaValueText(true);
        // Assistive technologies should make announcement for focus locked
        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.cell-focus-trapped' }));
        event.stopPropagation();
        event.preventDefault();
        break;
      case KeyCode.KEY_ESCAPE:
        // Assistive technologies should stop announcing aria-valueText once focus unlocked
        setIsAriaValueText(false);
        // Release focus lock
        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.resume-navigation' }));
        setNavigationEnabled(true);
        break;
      case KeyCode.KEY_RIGHT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Increase column width
          onResizeHandleChange(columnIndex, columnResizeIncrement);
        }
        break;
      case KeyCode.KEY_LEFT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Decrease column width
          onResizeHandleChange(columnIndex, -columnResizeIncrement);
        }
        break;
      default:
    }

    if (!isNavigationEnabled) {
      event.stopPropagation();
      event.preventDefault();
    }
  };

  // Prevent click event propagation
  const onClick = (event) => {
    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
  };

  const onBlur = () => {
    setNavigationEnabled(true);
    setIsAriaLabel(false);
    setIsActive(false);
  };

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/no-static-element-interactions
    <div
      ref={resizeHandleRef}
      draggable
      role={isNavigationEnabled ? null : 'slider'}
      tabIndex={-1}
      aria-hidden={!isActive}
      aria-valuemin={isActive ? minimumWidth : null}
      aria-valuenow={isActive ? columnWidth : null}
      aria-valuemax={isActive ? maximumWidth : null}
      aria-label={isAriaLabel ? intl.formatMessage({ id: 'Terra.worklist-data-grid.resize-handle-template' }, { columnText }) : null}
      aria-valuetext={isAriaValueText ? intl.formatMessage({ id: 'Terra.worklist-data-grid.resize-handle-value-text' }, { columnWidth }) : null}
      style={{ height: `${height}px` }} // eslint-disable-line react/forbid-dom-props
      onMouseDown={onMouseDown}
      onMouseUp={onMouseUp}
      onMouseEnter={fitToTable}
      onMouseLeave={onMouseLeave}
      onKeyDown={onKeyDown}
      onClick={onClick}
      onFocus={fitToTable}
      onBlur={onBlur}
      className={cx('resize-handle', theme.className, { 'resize-handle-selected': !isNavigationEnabled })}
    />
  );
};

ColumnResizeHandle.propTypes = propTypes;
ColumnResizeHandle.defaultProps = defaultProps;
export default injectIntl(ColumnResizeHandle);
