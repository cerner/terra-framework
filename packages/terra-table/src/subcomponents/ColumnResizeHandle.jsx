import React, {
  useContext, useRef, useState, useEffect,
} from 'react';
import classNames from 'classnames/bind';
import { injectIntl } from 'react-intl';
import * as KeyCode from 'keycode-js';
import PropTypes from 'prop-types';

import ThemeContext from 'terra-theme-context';

import ColumnContext from '../utils/ColumnContext';
import GridContext, { GridConstants } from '../utils/GridContext';
import styles from './ColumnResizeHandle.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  id: PropTypes.string,
  /**
   * The cell's column position in the grid. This is zero based.
   */
  columnIndex: PropTypes.number,

  /**
   * Text of the column associated with the divider.
   */
  columnText: PropTypes.string.isRequired,

  /**
   * Number that specifies the Width of the associated column in pixels.
   */
  columnWidth: PropTypes.number.isRequired,

  /**
   * Number that specifies the height of the resize handle in pixels.
   */
  height: PropTypes.number.isRequired,

  /**
   * String that specifies the init height for the resize handler to accomodate actions row.
   */
  resizeHandlerInitHeight: PropTypes.string,

  /**
   * Numeric increment in pixels to adjust column width when resizing via the keyboard.
   */
  columnResizeIncrement: PropTypes.number,

  /**
   * Control is the active element
   */
  isActive: PropTypes.bool,

  /**
   * Handler function to update isActive for parent.
   */
  setIsActive: PropTypes.func,

  /**
   * Number that specifies the minimum column width in pixels.
   */
  minimumWidth: PropTypes.number.isRequired,

  /**
   * Number that specifies the maximum column width in pixels.
   */
  maximumWidth: PropTypes.number.isRequired,

  /**
   * Function that is called when onMouseDown event is triggered for the resize handle.
   */
  onResizeMouseDown: PropTypes.func.isRequired,

  /**
   * Function that is called when onMouseDown event is triggered for the resize handle.
   */
  onResizeMouseUp: PropTypes.func.isRequired,
  /**
     * Function that is called when the the keyboard is used to adjust the column size.
     */
  onResizeHandleChange: PropTypes.func,

  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
};

const defaultProps = {
  columnResizeIncrement: 10,
};

const ColumnResizeHandle = (props) => {
  const {
    id,
    columnIndex,
    columnResizeIncrement,
    columnText,
    columnWidth,
    height,
    intl,
    isActive,
    maximumWidth,
    minimumWidth,
    resizeHandlerInitHeight,
    onResizeHandleChange,
    onResizeMouseDown,
    onResizeMouseUp,
    setIsActive,
  } = props;

  const theme = useContext(ThemeContext);
  const gridContext = useContext(GridContext);

  const columnContext = useContext(ColumnContext);
  const isGridContext = gridContext.role === GridConstants.GRID;

  // Ref variable for native resize handle element
  const resizeHandleRef = useRef();

  const [isNavigationEnabled, setNavigationEnabled] = useState(true);

  useEffect(() => {
    if (isActive) {
      resizeHandleRef.current.focus();
    }
  }, [isActive]);

  const onMouseDown = (event) => {
    // Set focus to resize handle DOM element
    resizeHandleRef.current.focus();

    onResizeMouseDown(event);

    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
    // Prevent default dragging behavior
    event.preventDefault();
  };

  const onMouseUp = () => {
    onResizeMouseUp();
  };

  const fitToTable = () => {
    // Update resize handle height to match parent table height
    resizeHandleRef.current.style.height = `${height}px`;
  };

  const onMouseLeave = () => {
    if (document.activeElement !== resizeHandleRef.current) {
      resizeHandleRef.current.style.height = resizeHandlerInitHeight || '100%';
    }
  };

  // Handle column resize handle keyboard navigation
  const onKeyDown = (event) => {
    const key = event.keyCode;
    switch (key) {
      case KeyCode.KEY_SPACE:
      case KeyCode.KEY_RETURN:
        setNavigationEnabled(false);
        // Lock focus into component
        resizeHandleRef.current.focus();

        // Assistive technologies should make announcement for focus locked
        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.table.cell-focus-trapped' }));
        event.stopPropagation();
        event.preventDefault();
        break;
      case KeyCode.KEY_ESCAPE:
        // Release focus lock
        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.table.resume-navigation' }));
        setNavigationEnabled(true);
        break;
      case KeyCode.KEY_RIGHT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Increase column width
          onResizeHandleChange(columnIndex, columnResizeIncrement);
        }
        break;
      case KeyCode.KEY_LEFT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Decrease column width
          onResizeHandleChange(columnIndex, -columnResizeIncrement);
        }
        break;
      default:
    }

    if (!isNavigationEnabled) {
      event.stopPropagation();
      event.preventDefault();
    }
  };

  // Prevent click event propagation
  const onClick = (event) => {
    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
  };

  const onBlur = () => {
    setNavigationEnabled(true);
    setIsActive(false);
  };

  return (
  // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/no-static-element-interactions
    <div
      id={id}
      ref={resizeHandleRef}
      draggable
      role={isNavigationEnabled ? null : 'slider'}
      tabIndex={isGridContext ? -1 : 0}
      aria-hidden={isGridContext ? !isActive : false}
      aria-valuemin={isActive ? minimumWidth : null}
      aria-valuenow={isActive ? columnWidth : null}
      aria-valuemax={isActive ? maximumWidth : null}
      aria-label={isActive && isNavigationEnabled ? intl.formatMessage({ id: 'Terra.table.resize-handle-template' }, { columnText }) : null}
      aria-valuetext={!isNavigationEnabled ? intl.formatMessage({ id: 'Terra.table.resize-handle-value-text' }, { columnWidth }) : null}
      // eslint-disable-next-line react/forbid-dom-props
      style={{ height: resizeHandlerInitHeight || '100%' }}
      onMouseDown={onMouseDown}
      onMouseUp={onMouseUp}
      onMouseEnter={fitToTable}
      onMouseLeave={onMouseLeave}
      onKeyDown={onKeyDown}
      onClick={onClick}
      onFocus={fitToTable}
      onBlur={onBlur}
      className={cx('resize-handle', theme.className, { 'resize-handle-selected': !isNavigationEnabled })}
    />
  );
};

ColumnResizeHandle.propTypes = propTypes;
ColumnResizeHandle.defaultProps = defaultProps;
export default injectIntl(ColumnResizeHandle);
