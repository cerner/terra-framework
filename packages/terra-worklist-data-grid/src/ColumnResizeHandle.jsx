import React, { useContext, useRef, useCallback } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './ColumnResizeHandle.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  active: PropTypes.bool,
  /**
   * Text of the column associated with the divider
   */
  columnText: PropTypes.string.isRequired,
  /**
   * Width of the column associated with the divider
   */
  columnWidth: PropTypes.number.isRequired,
  /**
   * String that specifies the default width for columns in the grid. Any valid CSS width value is accepted.
   */
  height: PropTypes.number.isRequired,
  /**
   * String that specifies the minimum column width in pixels
   */
  minimumWidth: PropTypes.number.isRequired,
  /**
   * String that specifies the minimum column width in pixels
   */
  maximumWidth: PropTypes.number.isRequired,
  /**
   * Function that is called when a selectable header cell is selected. Parameters: `onColumnSelect(columnId)`
   */
  onResizeMouseDown: PropTypes.func.isRequired,
};

const ColumnResizeHandle = (props) => {
  const {
    active,
    columnText,
    columnWidth,
    height,
    minimumWidth,
    maximumWidth,
    onResizeMouseDown,
  } = props;

  // Retrieve current theme from context
  const theme = useContext(ThemeContext);
  // Ref variable for native resize handle element
  const resizeHandle = useRef();

  // Ref callback to obtain the native resize handle element
  const resizeHandleRef = useCallback((node) => {
    resizeHandle.current = node;
  }, []);

  // Mouse down event listener to give focus to resize handler and notify the provider
  const onMouseDown = (event) => {
    resizeHandle.current.tabIndex = 0;
    resizeHandle.current.focus();

    onResizeMouseDown(event);

    event.stopPropagation();
    event.preventDefault();
  };

  const onClick = (event) => {
    event.stopPropagation();
    event.preventDefault();
  };

  return (
  /* eslint-disable react/forbid-dom-props */
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/interactive-supports-focus
    <div
      ref={resizeHandleRef}
      draggable
      role="slider"
      aria-valuemin={minimumWidth}
      aria-valuenow={columnWidth}
      aria-valuemax={maximumWidth}
      aria-label={columnText}
      aria-valuetext={`${columnWidth} pixels column width`}
      style={{ height: `${height}px` }}
      onMouseDown={onMouseDown}
      onClick={onClick}
      className={cx('resize-handle', theme.className, { active })}
    />
  );
};

ColumnResizeHandle.propTypes = propTypes;
export default injectIntl(ColumnResizeHandle);
