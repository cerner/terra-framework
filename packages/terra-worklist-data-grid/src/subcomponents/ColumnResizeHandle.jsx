import React, {
  useContext, useRef, useEffect, useState,
} from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import * as KeyCode from 'keycode-js';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import '../_elementPolyfill';
import styles from './ColumnResizeHandle.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The cell's column position in the grid. This is zero based.
   */
  columnIndex: PropTypes.number,
  /**
   * Text of the column associated with the divider.
   */
  columnText: PropTypes.string.isRequired,
  /**
   * Number that specifies the Width of the associated column in pixels.
   */
  columnWidth: PropTypes.number.isRequired,
  /**
   * Numeric increment in pixels to adjust column width when resizing via the keyboard
   */
  columnResizeIncrement: PropTypes.number,
  /**
   * Control is the active element
   */
  isActive: PropTypes.bool,
  /**
   * Number that specifies the minimum column width in pixels.
   */
  minimumWidth: PropTypes.number.isRequired,
  /**
   * Number that specifies the maximum column width in pixels.
   */
  maximumWidth: PropTypes.number.isRequired,
  /**
   * Function that is called when onMouseDown event is triggered for the resize handle
   */
  onResizeMouseDown: PropTypes.func.isRequired,
  /**
   * Function that is called when the the keyboard is used to adjust the column size
   */
  onResizeHandleChange: PropTypes.func,
  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
};

const defaultProps = {
  columnResizeIncrement: 10,
};

const ColumnResizeHandle = (props) => {
  const {
    columnIndex,
    columnText,
    columnWidth,
    columnResizeIncrement,
    isActive,
    minimumWidth,
    maximumWidth,
    onResizeMouseDown,
    onResizeHandleChange,
    intl,
  } = props;

  // Retrieve current theme from context
  const theme = useContext(ThemeContext);
  // Ref variable for native resize handle element
  const resizeHandleRef = useRef();

  const [isNavigationEnabled, setNavigationEnabled] = useState(true);

  useEffect(() => {
    if (isActive) {
      resizeHandleRef.current.focus();
    } else {
      resizeHandleRef.current.blur();
    }
  }, [isActive]);

  // Mouse down event listener to give focus to resize handler and notify the provider
  const onMouseDown = (event) => {
    // Set focus to resize handle DOM element
    resizeHandleRef.current.focus();

    // Execute callback function to notify consumer of mouse down event
    onResizeMouseDown(event);

    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
    // Prevent default dragging behavior
    event.preventDefault();
  };

  const fitToTable = () => {
    // Find parent table element
    const parentTable = resizeHandleRef.current.closest('table');

    // Update resize handle height to match parent table height
    if (parentTable) {
      resizeHandleRef.current.style.height = `${parentTable.offsetHeight}px`;
    }
  };

  const onMouseLeave = () => {
    if (document.activeElement !== resizeHandleRef.current) {
      resizeHandleRef.current.style.height = '100%';
    }
  };

  // Handle column resize handle keyboard navigation
  const onKeyDown = (event) => {
    const key = event.keyCode;
    switch (key) {
      case KeyCode.KEY_SPACE:
      case KeyCode.KEY_RETURN:
        // Lock focus into component
        resizeHandleRef.current.focus();
        setNavigationEnabled(false);
        event.stopPropagation();
        event.preventDefault();
        break;
      case KeyCode.KEY_ESCAPE:
        // Release focus lock
        setNavigationEnabled(true);
        break;
      case KeyCode.KEY_RIGHT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Increase column width
          onResizeHandleChange(columnIndex, columnResizeIncrement);
        }
        break;
      case KeyCode.KEY_LEFT:
        if (onResizeHandleChange && !isNavigationEnabled) {
          // Decrease column width
          onResizeHandleChange(columnIndex, -columnResizeIncrement);
        }
        break;
      default:
    }

    if (!isNavigationEnabled) {
      event.stopPropagation();
      event.preventDefault();
    }
  };

  // Prevent click event propagation
  const onClick = (event) => {
    // Prevent event bubbling since necessary actions are handled by this component
    event.stopPropagation();
  };

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
    <div
      ref={resizeHandleRef}
      draggable
      role="slider"
      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
      tabIndex={isActive ? 0 : -1}
      aria-hidden={!isActive}
      aria-valuemin={minimumWidth}
      aria-valuenow={columnWidth}
      aria-valuemax={maximumWidth}
      aria-label={intl.formatMessage({ id: 'Terra.worklist-data-grid.resize-handle-template' }, { columnText })}
      // aria-valuetext={intl.formatMessage({ id: 'Terra.worklist-data-grid.resize-handle-template' }, { columnText })}
      onMouseDown={onMouseDown}
      onMouseEnter={fitToTable}
      onMouseLeave={onMouseLeave}
      onKeyDown={onKeyDown}
      onClick={onClick}
      onFocus={fitToTable}
      onBlur={() => { setNavigationEnabled(true); }}
      className={cx('resize-handle', theme.className)}
    />
  );
};

ColumnResizeHandle.propTypes = propTypes;
ColumnResizeHandle.defaultProps = defaultProps;
export default injectIntl(ColumnResizeHandle);
