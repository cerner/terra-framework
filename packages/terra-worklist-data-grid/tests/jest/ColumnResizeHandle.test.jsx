/* eslint-disable import/no-extraneous-dependencies */
// eslint-disable no-console
import React from 'react';
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import ColumnResizeHandle from '../../src/ColumnResizeHandle';

describe('ColumnResizeHandle', () => {
  it('render a default column resize handle', () => {
    const mockResizeMouseDown = jest.fn();

    const wrapper = mountWithIntl(
      <ColumnResizeHandle
        columnIndex={0}
        columnText="Vitals"
        columnWidth={200}
        height="500px"
        minimumWidth={60}
        maximumWidth={300}
        onResizeMouseDown={mockResizeMouseDown}
      />,
    );

    const resizeHandle = wrapper.find('.resize-handle:not(.active)');
    expect(resizeHandle).toHaveLength(1);
    expect(resizeHandle.props()['aria-valuetext']).toBe('200 pixels column width');
    expect(resizeHandle.props()['aria-label']).toBe('Vitals');
    expect(resizeHandle.props()['aria-valuenow']).toBe(200);
    expect(resizeHandle.props()['aria-valuemin']).toBe(60);
    expect(resizeHandle.props()['aria-valuemax']).toBe(300);
    expect(resizeHandle.props().onClick).toBeDefined();
    expect(resizeHandle.props().style.height).toBe('500px');

    resizeHandle.invoke('onMouseDown')(
      {
        stopPropagation: () => {},
        preventDefault: () => {},
      },
      9000,
    );

    expect(mockResizeMouseDown).toHaveBeenCalled();

    expect(wrapper).toMatchSnapshot();
  });

  it('render an active column resize handle', () => {
    const mockResizeMouseDown = jest.fn();

    const wrapper = shallowWithIntl(
      <ColumnResizeHandle
        active
        columnIndex={0}
        columnText="Vitals"
        columnWidth={200}
        height="500px"
        minimumWidth={60}
        maximumWidth={300}
        onResizeMouseDown={mockResizeMouseDown}
      />,
    ).dive();

    const resizeHandle = wrapper.find('.resize-handle.active');
    expect(resizeHandle).toHaveLength(1);
    expect(resizeHandle.props()['aria-valuetext']).toBe('200 pixels column width');
    expect(resizeHandle.props()['aria-label']).toBe('Vitals');
    expect(resizeHandle.props()['aria-valuenow']).toBe(200);
    expect(resizeHandle.props()['aria-valuemin']).toBe(60);
    expect(resizeHandle.props()['aria-valuemax']).toBe(300);
    expect(resizeHandle.props().onClick).toBeDefined();
    expect(resizeHandle.props().style.height).toBe('500px');

    expect(wrapper).toMatchSnapshot();
  });
});
