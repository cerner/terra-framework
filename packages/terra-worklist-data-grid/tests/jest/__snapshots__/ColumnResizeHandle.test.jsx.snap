// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnResizeHandle render a default column resize handle 1`] = `
<InjectIntl(ColumnResizeHandle)
  columnIndex={0}
  columnText="Vitals"
  columnWidth={200}
  height={500}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  maximumWidth={300}
  minimumWidth={60}
  onResizeMouseDown={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "preventDefault": [Function],
            "stopPropagation": [Function],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <ColumnResizeHandle
    columnIndex={0}
    columnText="Vitals"
    columnWidth={200}
    height={500}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    maximumWidth={300}
    minimumWidth={60}
    onResizeMouseDown={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "preventDefault": [Function],
              "stopPropagation": [Function],
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <div
      aria-label="Vitals"
      aria-valuemax={300}
      aria-valuemin={60}
      aria-valuenow={200}
      aria-valuetext="200 pixels column width"
      className="resize-handle"
      draggable={true}
      onClick={[Function]}
      onMouseDown={[Function]}
      role="slider"
      style={
        Object {
          "height": "500px",
        }
      }
    />
  </ColumnResizeHandle>
</InjectIntl(ColumnResizeHandle)>
`;

exports[`ColumnResizeHandle render an active column resize handle 1`] = `
<div
  aria-label="Vitals"
  aria-valuemax={300}
  aria-valuemin={60}
  aria-valuenow={200}
  aria-valuetext="200 pixels column width"
  className="resize-handle active"
  draggable={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  role="slider"
  style={
    Object {
      "height": "500px",
    }
  }
/>
`;
